src/Microsoft.MobileBlazorBindings.WebView/BlazorHybridHost.cs

```diff
@@ -36,22 +45,78 @@ public static void AddResourceAssembly(Assembly resourceAssembly, string content
         }
 
         /// <summary>
-        /// Tries to load the given <paramref name="filename"/> from a registered resource assembly.
-        /// Resource assemblies are registered by calling <see cref="AddResourceAssembly(Assembly, string)"/>.
+        /// Adds the specified resources to the list of assemblies that contain static web resources for the
+        /// web portion of the application. The resources must be embedded in an assembly that directly
+        /// references the <c>Microsoft.Extensions.FileProviders.Embedded</c> NuGet package and specifies
+        /// <c>&lt;GenerateEmbeddedFilesManifest&gt;true&lt;/GenerateEmbeddedFilesManifest&gt;</c>
+        /// in the project settings.
+        /// </summary>
+        /// <param name="contentRoot">The root folder of the content in the application. This is prepended to filenames to get the full path.</param>
+        public static void AddDirectory(string contentRoot)
+        {
+            if (Directory.Exists(contentRoot))
+            {
+                var dirInfo = new DirectoryInfo(contentRoot);
+                contentRoot = dirInfo.FullName;
+
+                _resourceFileProviders.Add(
+                    new Lazy<IFileProvider>(
+                        valueFactory: () => new PhysicalFileProvider(contentRoot, Extensions.FileProviders.Physical.ExclusionFilters.Sensitive),
+                        isThreadSafe: true));
+            }
+            else
+            {
+                foreach (var lazyProvider in _resourceFileProviders)
+                {
+                    var provider = lazyProvider.Value;
+
+                    try
+                    {
+                        var fileInfo = provider.GetDirectoryContents("");
+
+                        if (fileInfo != null && fileInfo.Any() && fileInfo.First().PhysicalPath.Contains(contentRoot))
```

Comment:
Should it be `.StartsWith()` or `.Contains()`? (And with the appropriate CultureInfo/StringComparison?)